Code/Framework/AzFramework/Platform/Linux/AzFramework/Process/ProcessWatcher_Linux.cpp

```diff
@@ -279,6 +279,32 @@ namespace AzFramework
             }
             environmentVariables[numEnvironmentVars] = nullptr;
         }
+        else
+        {
+            // If no environment variables were specified, then copy the current process's environment variables
+            // and pass it along for the unwatch process.
+            extern char **environ;  // Read the current environment global system pointer
+            char **env = ::environ;
+            AZ_Assert(env, "Environment variables for current process not available\n");
+
+            // Do a first pass across the global ::environ array to count how many environment variables are set (the array is null-terminated)
+            for (;env[numEnvironmentVars]!=nullptr;numEnvironmentVars++) 
+            {
+            }
+
+            if (numEnvironmentVars>0)
+            {
+                environmentVariables = new char*[numEnvironmentVars + 1];
+                for (size_t index=0; index<numEnvironmentVars; index++)
```

Comment:
The whitespacing is off here.

Also I remember providing a code snippet for this to someone that didn't using explicit `new`

```suggestion
            // Do a first pass across the global ::environ array to count how many environment variables are set (the array is null-terminated)
            for (; env[numEnvironmentVars] != nullptr; numEnvironmentVars++) 
            {
            }

            if (numEnvironmentVars > 0)
            {
                environmentVariables = new char* [numEnvironmentVars + 1];
                for (size_t index=0; index < numEnvironmentVars; index++)
```

Reply:
`AZStd::vector<AZStd::unique_ptr<char[]>> environmentVariables` can be used here.

Then a call to `environmentVariables.data()` can be used

Reply:
will update, but will need to also update the section above as well

Reply:
@lumberyard-employee-dm: This is something we do in ProcessWatcher_Mac.cpp as well. https://github.com/o3de/o3de/blob/9d2352c3b74e3cf03127b09a184c66ad9d0b1170/Code/Framework/AzFramework/Platform/Mac/AzFramework/Process/ProcessWatcher_Mac.cpp#L279-L293

Reply:
Looking at the mac implementation, we probably don't need the string allocation and copy for the ::environ variables, and they can be passed down directly. Will update that

Reply:
Ahh, that is where that code was at.
Thanks @amzn-sj